test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -312,30 +312,115 @@ public void testUnknownField() throws IOException {
     }
 
     /**
-     * Test that adding additional object into otherwise correct query string
-     * should always trigger some kind of Parsing Exception.
+     * Test that adding an additional object within each object of the otherwise correct query always triggers some kind of
+     * parse exception. Some specific objects do not cause any exception as they can hold arbitray content; they can be
+     * declared by overriding {@link #getObjectsHoldingArbitraryContent()}
      */
-    public void testUnknownObjectException() throws IOException {
+    public final void testUnknownObjectException() throws IOException {
         String validQuery = createTestQueryBuilder().toString();
+        unknownObjectExceptionTest(validQuery);
+        for (String query : getAlternateVersions().keySet()) {
+            unknownObjectExceptionTest(query);
+        }
+    }
+
+    private void unknownObjectExceptionTest(String validQuery) throws IOException {
```

Comment:
Can we add the general idea about how this test is supposed to modify the original query here, maybe with a small example? e.g. we just talked about that the outer loop that traverses the whole json query  adds a new object within each object (after encountering an opening bracket, under certain conditions) of the whole big query. This doesn't need to account for all kinds of exceptions (like we are inside quotes etc...)